tests/conftest.py

```diff
@@ -202,3 +203,19 @@ def _call(*args, **kwargs):
         return calls
 
     return _capcall
+
+
+@pytest.fixture
+def set_timezone(monkeypatch):
+    """Fixture to temporarily set a timezone.
+
+    Will use a very implausible timezone if none is provided.
+    """
+    def patch(time_zone: str = 'XXX-19:17'):
+        monkeypatch.setenv('TZ', time_zone)
+        time.tzset()
+
+    try:
+        yield patch
+    finally:
+        time.tzset()  # Reset to the original time zone after the test
```

Comment:
Argh, unfortunately I have debugged this and found that the TZ env var is remaining set when `tzset()` is called in the `finally` block, so this does not reset the TZ on Linux.

However, the solution is simple:
```suggestion
        monkeypatch.undo()
        time.tzset()  # Reset to the original time zone after the test
```